cmake/ConfigSGX.cmake

```diff
@@ -10,27 +10,27 @@ execute_process(COMMAND getconf LONG_BIT OUTPUT_VARIABLE VAR_LONG_BIT)
 # set them to the usual defaults.
 set(SGX_SDK "$ENV{SGX_SDK}")
 if("${SGX_SDK} " STREQUAL " ")
-       set(SGX_SDK /opt/intel/sgxsdk)
+    message(FATAL_ERROR "SGX_SDK: undefined environment variable")
 endif()
 
 set(SGX_ARCH "$ENV{SGX_ARCH}")
 if("${SGX_ARCH} " STREQUAL " ")
-       set(SGX_ARCH x64)
+    message(FATAL_ERROR "SGX_ARCH: undefined environment variable")
 endif()
 
 set(SGX_MODE "$ENV{SGX_MODE}")
 if("${SGX_MODE} " STREQUAL " ")
-       set(SGX_MODE HW) # SGX mode: sim, hw
+    message(FATAL_ERROR "SGX_MODE: undefined environment variable")
 endif()
 
 set(SGX_BUILD "$ENV{SGX_BUILD}")
 if("${SGX_BUILD} " STREQUAL " ")
-       set(SGX_BUILD PRERELEASE)
+    message(FATAL_ERROR "SGX_BUILD: undefined environment variable")
 endif()
 
 set(SGX_SSL "$ENV{SGX_SSL}")
 if("${SGX_SSL} " STREQUAL " ")
-       set(SGX_SSL /opt/intel/sgxssl)
+    message(FATAL_ERROR "SGX_SSL: undefined environment variable")
 endif()
```

Comment:
I like these changes. Making it fatal might be something to do in the toplevel file depending on where it's used.

Reply:
I am not following. This file is supposed to be a toplevel one that should be include in cmake-based sgx builds.

Reply:
is regular make not ever involved in the non-cmake builds?

Reply:
it is. Actually the build starts with the regular make, which includes `config.mk` (where the variables are defined). Then make calls cmake, which inherits these definitions. The first thing  cmake files do (in sgx builds) is to include this `ConfigSGX.cmake`, which checks for the existence of the required definitions.